CabalHelper/Compile.hs

```diff
@@ -413,6 +413,9 @@ errorInstallCabal cabalVer _distdir = panic $ printf "\
  where
    sver = showVersion cabalVer
 
+helperPrepared :: Version -> IO Bool
+helperPrepared compCabalVersion = isJust <$> cachedExe compCabalVersion
```

Comment:
Yeah when querying the Cabal source package we need to compile the helper with the very same source because Cabal's `Setup.hs` actually uses the modules from the source tree instead of the Cabal library from the package-db.

I remember talking about this, just couldn't recall the details or find the discussion again just then. Still we need to think about how the IDE would even handle the case I outlined above. If I understand correctly the only thing the IDE could do when `is-prepared` always returns false is to keep annoying the user about the fact that cabal-helper will recompile itself.